Client/Frontend/Browser/BrowserLongPressGestureRecognizer.swift

```diff
@@ -0,0 +1,135 @@
+    }
+
+    /// Recursively call block on view and its subviews
+    private func recursiveBlockOnViewAndSubviews(mainView: UIView, block:(view: UIView) -> Void) {
+        block(view: mainView)
+        mainView.subviews.map(){ self.recursiveBlockOnViewAndSubviews($0 as UIView, block) }
+    }
+
+    /// Find location in screen corresponding to webview - in case it is zoomed or scrolled
+    private func rectLocationInWebView(webView:WKWebView,locationRect:CGRect) -> CGRect {
+        var rect = locationRect
+        var scale = self.browser.webView.scrollView.zoomScale
+        rect.origin.x *= scale
+        rect.origin.y *= scale
+        rect.size.width *= scale
+        rect.size.height *= scale
+        rect.origin.x += self.browser.webView.scrollView.contentInset.left;
+        rect.origin.y += self.browser.webView.scrollView.contentInset.top;
+
+        return rect
+    }
+
+    // MARK: - BrowserHelper Mehods
+    class func name() -> String {
+        return "BrowserLongPressGestureRecognizer"
+    }
+
+    func scriptMessageHandlerName() -> String? {
+        return "longPressMessageHandler"
+    }
+
+    func userContentController(userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        var elementsDict: [String:AnyObject]? = message.body as? [String:AnyObject]
+        if elementsDict  == nil {
+            return
+        }
+
+        var elements:[ElementType] = []
+        if let hrefElement = elementsDict!["hrefElement"] as? [String:String] {
+            if let hrefStr: String = hrefElement["hrefLink"] {
+                if let hrefLinkURL = NSURL(string: hrefStr) {
+                    elements.append(ElementType.hrefLink(hrefLinkURL))
+                }
+            }
+        }
+
+        if let imageElement = elementsDict!["imageElement"] as? [String:String] {
+            if let imageSrcStr: String = imageElement["imageSrc"] {
+                if let imageSrc = NSURL(string: imageSrcStr) {
+                    elements.append(ElementType.ImageSrc(imageSrc))
+                }
+            }
+        }
+        var disableGestures:[UIGestureRecognizer] = []
+        self.recursiveBlockOnViewAndSubviews(self.browser.webView) { view in
+            if let gestureRecognizers = view.gestureRecognizers as? [UIGestureRecognizer] {
+                for g in gestureRecognizers {
+                    if g != self  && g.enabled == true {
+                        g.enabled = false
+                        disableGestures.append(g)
+                    }
+                }
+            }
+        }
+        if elements.count > 0 {
+            self.browserLongPressGestureDelegate?.browser(browser, elementsDidLongPressed: elements)
+        }
+        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(0.5 * Double(NSEC_PER_SEC))), dispatch_get_main_queue()) {
```

Comment:
I still strongly dislike doing this, and I'm not even sure you need it. Testing this PR locally, I removed this dispatch and called <code>disableGestures.map({ $0.enabled = true })</code> immediately after calling the delegate above. It still seems to work fine -- so can we remove this delayed dispatch altogether?